DIPs/DIP1001.md

```diff
@@ -0,0 +1,291 @@
+generate C style error code handling for that function using a tagged union of
+the result and all __ErrorCode exceptions.
+
+### Rationale
+
+#### Throws Property
+
+Main Reason to include this is so Throw Any Type and Error Code Unwinding
+work with incremental compilation.
+
+Aids maintance, callers of functions can know what to catch.
+
+##### Vs C++ throws lists
+
+Due to C++'s header files and legacy ploblem, exceptions had to be checked at runtime,
+This is not the case with This proposal's @throws attribute, they can be checked at
+compile time.
+
+Unlike C++'s throws lists, @throws is usable in templates because they will be applied
+automatically.
+
+@throws being a attribute, may be applied to multiple functions at once.
+
+#### Throw Any Type
+
+Aids performace, doesn't force the exception to generate a stack trace or message.
+
+#### Error Code Unwinding
+
+Allow users to choose between the error code style of exception
+handling or no cost tables.
+
+The @throws() attribute is necessary for incremental compilation.
+
+### Breaking changes / deprecation process
+
+None
+
+### Examples
+
+Throw By Value:
+
+	struct ThrowByValue
+	{
+		int[4] filler;
+	}
+
+	int throwValue() @throws(ThrowByValue)
+	{
+		throw ThrowByValue([0, 1, 2, 3]);
+	}
+
+	void catchValue()
+	{
+		try
+		{
+		    throwValue();
+		}
+		catch (ThrowByValue v)
+		{
+
+		}
+	}
+
+C Style Exception Handling:
+
+	struct CStyle
+	{
+		enum __ErrorCode = true;
+		int[4] filler;
+	}
+
+	int throwUp() @throws(CStyle)
+	{
+		throw CStyle();
+	}
+
+	int cleanUp()
+	{
+		try
+		{
+		    return throwUp();
+		}
+		catch (CStyle code)
+		{
+		    return 0;
+		}
+	}
+Should compile into something like this:
+
+	struct CStyle
+	{
+		enum __ErrorCode = true;
+		int[4] filler;
+	}
+
+	struct throwUpResult
+	{
+		size_t id;
+		union
+		{
+		    int regular;
+		    CStyle exception;
+		}
+	}
+
+	throwUpResult throwUp()
+	{
+		throwUpResult result;
+		result.id = 1;
+		result.exception = CStyle();
+		return result;
+	}
+
+	int cleanUp()
+	{
+		auto throwUpCheck = throwUp();
+		if (throwUpCheck.id == 0)
+		{
+		    return throwUp.regular;
+		}
+		else
+		{
+		    return 0;
+		}
+	}
+
+Mix and Match:
+
+	struct ThrowByValue
+	{
+		int[4] filler;
+	}
+
+	struct ErrorCode1
+	{
+		enum __ErrorCode = true;
+	}
+
+	struct ErrorCode2
+	{
+		enum __ErrorCode = true;
+	}
+
+	int mixAndMatch() @throws(ThrowByValue, ErrorCode1, ErrorCode2)
+	{
+		return 0;
+	}
+
+	/* should generate c style error code handling for ErrorCode1 and ErrorCode2
+	 * but not ThrowByValue
+	 */
+	void catchValue()
+	{
+		try
+		{
+		    mixAndMatch();
+		}
+		catch (ThrowByValue v)
+		{
+
+		}
+		catch (ErrorCode1 e)
+		{
+
+		}
+		catch (ErrorCode2 e)
+		{
+
+		}
+	}
+
+Implict Conversion:
+
+	void throwLong() @throws(long)
+	{
+		//implictly converts to long
+		throw 0;
+	}
+## Copyright & License
+
+Copyright (c) 2016 Freddy A Cubas "Superstar64"
```

Comment:
So we are GNUtards now ? The License allows the D Fundation to do what it wants with it, isn't it enough ?